### PR TITLE
Update .travis.yml

### PR DESCRIPTION
**Pull request by nageshlop**
This PR adds CI support for the IBM Power Little Endian (ppc64le) architecture. The idea is to ensure that the builds on this architecture are continuously tested along with the Intel builds (amd64) as this is part of the ubuntu distro on that architecture as well and detecting (and fixing) any issues or failures early would help to ensure that we are always up to date.
The build and test results are available at the below location.
https://travis-ci.com/github/nageshlop/faye-websocket-node

### CODE REVIEW

**Comment by jcoglan** [comment]
Hi @nageshlop, have you ever had a problem with this package on your architecture? If so, what was it and how was the problem addressed?

**Comment by nageshlop** [comment]
Hi @jcoglan 
I didn't face any issue while building and testing this package. The little endian support for the IBM Power Platform (ppc64le) was introduced in 2014 and has been gaining momentum since. The latest versions of all the major distros including ubuntu 20.x, RHEL 7/8, Centos 7/8, Fedora, Debian, Alpine have support for ppc64le and we have close to probably 1K communities running this on travis and other CIs. Our team at IBM has been working with several open source communities over the last 6+ years to enhance and increase the overall ecosystem of open source packages on Power. Besides, there are several enterprises who run their workloads on these servers today, hope that helps. 
Thank you.

**Comment by jcoglan** [comment]
Without any examples of how this has detected and resolved problems, or how it would benefit the majority of my users, I'm not willing to commit to this. All this does is move the responsibility for ensuring this package runs on IBM hardware -- it moves that responsibility from IBM, onto me. It requires that I accept a set of costs and responsibilities, at no benefit to myself or to most of the package's users, while IBM captures most of the upside in the form of reduced costs.

From my point of view as a volunteer maintainer, in essence I'd be doing free work for IBM, and I'm not willing to do that.

**Comment by nageshlop** [comment]
tagging @gerrith3.

**Comment by gerrith3** [comment]
@jcoglan We have run automation on 100k npm packages and found that somewhere between 20-25% of them fail when building/running on ppc64le.  All of these claim to be architecture independent, but the reality is that there are environmental factors that most contributors don't think about until they see a failure.  We have tested your package currently and it runs, but we are trying to ensure that we have full parity with Intel in the realm of npms and pretty much everything else.  We are trying to intelligently test and validate these packages as we encounter ones that are currently working, and also fixing others as we get to them.

In terms of support - if, as you claim, this package is completely architecture independent, then testing should never encounter an error and therefore the overhead is only the cost of running on Travis.  IBM has made available about $1M USD of hardware for free to Travis, specifically to suppor the open source communities.  This effectively works as a label that says that you not only claim to be architecture independent but can back it up with testing.  The testing shouldn't cost you anything (we already paid the bill) and we have tested up front to make sure your projects to build and pass their own tests on Power.  We also document any limitations, e.g. dependencies on versions of javascript, node.js, etc. which don't run on Power.
Basically, we are doing all of the work for you, paying for all of the testing, and, if there are no issues, there is no overhead.  If there are issues then someone is trying to check in something which is no longer architecture independent and you get notified (or more importantly, the contributor gets notified and can evaluate their own patch directly).

**Comment by jcoglan** [comment]
@gerrith3 As I said above, I am not willing to enter into the commitments this PR is implicitly placing on me, and I will not be discussing my decision further.

There is however one detail in your response that I feel is important to correct:

> as you claim, this package is completely architecture independent

**I do not, and have not, made any such claim**, either here or in the project's documentation, and it is incorrect to assert that I have done so. The only implied compatibility claim made by this project is that it runs on [node.js v0.8.0 or above](https://github.com/faye/faye-websocket-node/blob/0.11.3/package.json#L13).

The software's [license](https://github.com/faye/faye-websocket-node/blob/0.11.3/LICENSE.md) states:

> software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied

You must not make assumptions about compatibility that I have not committed to, and you must not assert that I have made such commitments. You should not rely on the project's source code hosting or continuous integration/testing processes to ensure your own compatibility goals are met.

If you find bugs in the software on your architecture then I may consider accepting patches, but I will not be doing your compatibility work for you.
